llvm/lib/Transforms/IPO/FunctionAttrs.cpp

```diff
@@ -118,7 +118,7 @@ static void addLocAccess(MemoryEffects &ME, const MemoryLocation &Loc,
   if (isNoModRef(MR))
     return;
 
-  const Value *UO = getUnderlyingObject(Loc.Ptr);
+  const Value *UO = getUnderlyingObjectAggressive(Loc.Ptr);
```

Comment:
Compile-time impact: http://llvm-compile-time-tracker.com/compare.php?from=c1b70fa5bfea973d4141e27cf9668e9325609e19&to=9c73bbf882a265931e4669e6da2a1c99acbb4efe&stat=instructions:u